index.js

```diff
@@ -51,6 +53,63 @@ function filter(obj,options) {
                 break;
             }
         }
+
+        for (let method of options.methods) {
+            if (key === method) {
+                if (options.inverse) {
+                    if (Array.isArray(obj)) {
+                        // we need to seed the presence of an empty array
+                        // otherwise jptr won't know whether it's setting
+                        // an array entry or a property with a numeric key #26
+                        const components = state.path.split('/');
+                        components.pop(); // throw away last item
+                        if (jptr(filtered,components.join('/')) === false) {
+                            jptr(filtered,components.join('/'),[]);
+                        }
+                    }
+                    jptr(filtered,state.path,clone(obj[key]));
+                }
+                filteredpaths.push(state.path);
+                delete obj[key];
+                break;
+            }
+        }
+
+        for (let operationId of options.operationIds) {
+            if (obj[key] === operationId) {
+                if (options.inverse) {
+                    if (Array.isArray(obj)) {
+                        // we need to seed the presence of an empty array
+                        // otherwise jptr won't know whether it's setting
+                        // an array entry or a property with a numeric key #26
+                        const components = state.path.split('/');
+                        components.pop(); // throw away last item
+                        if (jptr(filtered,components.join('/')) === false) {
+                            jptr(filtered,components.join('/'),[]);
+                        }
+                    }
+                    jptr(filtered,state.path,clone(obj[key]));
+                }
+
+                // Recursed to a level to deep, so parent state
```

Comment:
Not sure I understand this comment?

Reply:
The recursive goes to depth 3, so when deleting the obj[key], it would only remove the `operationId` and not the whole method.

Original state that matches:

```
state {
  parent: {
    get: {
      'x-tim': true,
      operationId: 'get-domain-clusters',
      tags: [Array],
      summary: 'Get all domain cluster configurations',
      responses: [Object],
      description: 'Returns a list of all domain cluster configurations, for all list types.\n' +
        'The configurations are returned sorted by name.'
    },
    post: {
      operationId: 'post-domain-clusters',
      tags: [Array],
      summary: 'Create a new domain cluster configuration',
      requestBody: [Object],
      responses: [Object]
    }
  },
  path: '#/paths/~1portal~1v1~1domainclusters/post/operationId',
  depth: 3,
  pkey: 'post',
  payload: {},
  seen: WeakMap { <items unknown> },
  identity: false,
  identityDetection: false,
  key: 'operationId',
  identityPath: undefined
}
state.path #/paths/~1portal~1v1~1domainclusters/post/operationId
```

But I want to remove the whole method (like below), which is already passed in the recursive loop

Wanted obj[key] & state to delete
```
state {
  parent: { '/portal/v1/domainclusters': { get: [Object], post: [Object] } },
  path: '#/paths/~1portal~1v1~1domainclusters/post',
  depth: 2,
  pkey: '/portal/v1/domainclusters',
  payload: {},
  seen: WeakMap { <items unknown> },
  identity: false,
  identityDetection: false,
  key: 'post',
  identityPath: undefined
}
state.path #/paths/~1portal~1v1~1domainclusters/post
```

So what I did (not perfect, I admit, but I didn't found an alternative):

- pop the last element from the component, to go up a depth level, extract that statepath and then recurse over the full src again, just to delete the parent level of my operationID.

```
// Recursed to a level to deep, so parent state
                const components = state.path.split('/');
                components.pop(); // throw away last item
                const parentstate = components.join('/')

                // Loop over src to remove the item attached to the parent state
                recurse(src,{},function(obj,key,state){
                    if (state.path == parentstate)
                    {
                        delete obj[key];
                    }
                });

```

If you have a cleaner way, let me know and I'll be glad to improve the piece of code

Reply:
Ok, it was the `to` instead of `too` which partially confused me, but also the dangling `so parent state`...

You should be able to remove the whole containing object by deleting `state.parent[state.pkey]`

Reply:
I think I corrected it using the `state.parent[state.pkey]`, which I never would have found out by myself. So thanks for the great explanation.

I think I got everything covered for this PR.

I'm going to open another topic to better understand the "valid" & "inverse" properties, since I think it is not yet properly supported for the newly added options.